_posts/2022-01-21-safe-releases.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+title: Reducing thrash and increasing throughput by focusing on elegant releases
+author_avatar: https://avatars.githubusercontent.com/u/8116516?v=4
+author_name: Xavier Hocquet
+author_github_username: xhocquet
+---
+
+Continuous integration and deployment is fundamental to a high-performing technology business as set forth by Kim et al. Deploying frequently results in smaller releases, more frequent updates for your users, quicker feedback from real users, and a tighter feedback loop between business decisions and shipped products.
+
+The worst enemy of continuous deployment is a broken build. The larger the engineering team, the worse it is. If a bad data migration goes out, all releases may have to be paused until a resolution is shipped. Paused releases mean that code branches pile up and inevitably time will be spent fixing issues or rebasing pull requests.
+
+While there are many tools at our disposal to solve some of these problems, I believe a solid understanding of the basics helps us reason about releases and avoid the biggest problems in the first place.
+
+Below I will outline the release cycle for many modern apps (with a focus on Heroku/AWS and Ruby/Javascript), as well as some guidelines to plan graceful releases and migrations.
```

Comment:
Might be cool to set up the structure of the post a bit here. I really liked how you outlined the "danger zone", then outlined how to avoid those danger zones, and gave a real-world example of doing so.

Maybe something like "Below I will outline some common pitfalls in release cycles, as well as some guidelines to avoid those pitfalls and plan graceful releases and migrations"? Just a thought.

Reply:
Great feedback!